staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -528,6 +528,11 @@ func completeOpenAPI(config *openapicommon.Config, version *version.Info) {
 	}
 }
 
+// StopNotify returns a lifecycle signal of genericapiserver shutting down.
+func (c *Config) StopNotify() <-chan struct{} {
+	return c.lifecycleSignals.ShutdownInitiated.Signaled()
```

Comment:
this will panic if lifecycleSignals was not initialized , are we fine with that?

Reply:
should we check and initialize if not set?
```
if c.lifecycleSignals == nil {
 c.lifecycleSignals := newLifecycleSignals()
}
return c.lifecycleSignals.ShutdownInitiated.Signaled()
```

Reply:
If you're using NewConfig() [which is used across the codebase] lifecycleSignals will always be set.
If you're constructing the config yourself and not initialize lifecycleSignals, we will panic in other places anyway, e.g.:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go#L401

So I would say it's fine.

Reply:
In fact it's not a pointer - so we can't even compare it with nil. Just individual fields may be uninitialized...